pkg/agent/route/route_windows.go

```diff
@@ -100,21 +111,42 @@ func (c *Client) Reconcile(podCIDRs []string) error {
 // It overrides the routes if they already exist, without error.
 func (c *Client) AddRoutes(podCIDR *net.IPNet, nodeName string, peerNodeIP, peerGwIP net.IP) error {
 	obj, found := c.hostRoutes.Load(podCIDR.String())
+	route := &netroute.Route{
+		DestinationSubnet: podCIDR,
+	}
+	if c.networkConfig.TrafficEncapMode.NeedsEncapToPeer(peerNodeIP, c.nodeConfig.NodeIPAddr) {
+		route.LinkIndex = c.nodeConfig.GatewayConfig.LinkIndex
+		route.GatewayAddress = peerGwIP
+	} else if !c.networkConfig.TrafficEncapMode.NeedsRoutingToPeer(peerNodeIP, c.nodeConfig.NodeIPAddr) {
+		// NoEncap traffic to Node on the same subnet.
+		// Set the peerNodeIP as next hop.
+		route.LinkIndex = c.bridgeInfIndex
```

Comment:
Question - why this is needed?

Reply:
`route.LinkIndex` (br-int's index) is needed for `NewNetRoute` or `RemoveNetRoute`.